### PR TITLE
Github issue #1030 product info fix

### PR DESCRIPTION
**Pull request by jebrando**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/inc/internal/iothubtransport_amqp_device.h [hunk 1]
```diff
@@ -93,7 +93,8 @@ typedef struct DEVICE_CONFIG_TAG
 {
     const char* device_id;
     const char* module_id;
-    char* product_info;
+    pfTransport_GetOption_Product_Info_Callback prod_info_cb;
```
Comment (massand) [inline iothub_client/inc/internal/iothubtransport_amqp_device.h]: >pfTransport_GetOption_Product_Info_Callback [](start = 4, length = 43)

Please follow naming conventions of callbacks (such as the typedefs above)

File: iothub_client/inc/internal/iothubtransport_amqp_messenger.h [hunk 2]
```diff
@@ -106,6 +106,9 @@ typedef struct AMQP_MESSENGER_CONFIG_TAG
 
     AMQP_MESSENGER_SUBSCRIPTION_CALLBACK on_subscription_changed_callback;
     void* on_subscription_changed_context;
+
+    pfTransport_GetOption_Product_Info_Callback prod_info_cb;
```
Comment (massand) [inline iothub_client/inc/internal/iothubtransport_amqp_messenger.h]: >pfTransport_GetOption_Product_Info_Callback prod_info_cb; [](start = 4, length = 57)

Having 2 additional pointers in each header is unnecessary if we pass in the AMQP_DEVICE_HANDLE itself.

Comment (jebrando) [reply iothub_client/inc/internal/iothubtransport_amqp_messenger.h]: Maybe in the future we will consolidate some things, but the device handle is not used everywhere.

---
In reply to: [295569174](https://github.com/Azure/azure-iot-sdk-c/pull/1056#discussion_r295569174) [](ancestors = 295569174)

**Pull request merged**